plugins/importexport/doaj/DOAJPlugin.inc.php

```diff
@@ -103,6 +144,326 @@ function exportJournal(&$journal, $outputFile = null) {
+
+		// Return articles from published issues only.
+		$articles = array();
+		while ($article = $articleIterator->next()) {
+			$articles[] = $this->_prepareArticleData($article, $journal);
+			unset($article);
+		}
+		unset($articleIterator);
+
+		return $articles;
+	}
+
+	/**
+	 * Return the issue of an article.
+	 *
+	 * The issue will be cached if it is not yet cached.
+	 *
+	 * @param $article Article
+	 * @param $journal Journal
+	 *
+	 * @return Issue
+	 */
+	function _getArticleIssue($article, $journal) {
+		$issueId = $article->getIssueId();
+
+		// Retrieve issue if not yet cached.
+		$cache = $this->_getCache();
+		if (!$cache->isCached('issues', $issueId)) {
+			$issueDao = DAORegistry::getDAO('IssueDAO'); /* @var $issueDao IssueDAO */
+			$issue = $issueDao->getIssueById($issueId, $journal->getId(), true);
+			assert(is_a($issue, 'Issue'));
+			$nullVar = null;
+			$cache->add($issue, $nullVar);
+			unset($issue);
+		}
+
+		return $cache->get('issues', $issueId);
+	}
+
+	/**
+	 * Retrieve all articles for the given issue
+	 * and commit them to the cache.
+	 * @param $issue Issue
+	 * @return array
+	 */
+	function _retrieveArticlesByIssueId($issueId) {
+		$articlesByIssue = array();
+		$cache = $this->_getCache();
+
+		if (!$cache->isCached('articlesByIssue', $issueId)) {
+			$articleDao = DAORegistry::getDAO('PublishedArticleDAO');
+			$articles = $articleDao->getPublishedArticles($issueId);
+			if (!empty($articles)) {
+				foreach ($articles as $article) {
+					$cache->add($article, $nullVar);
+					unset($article);
+				}
+				$cache->markComplete('articlesByIssue', $issueId);
+				$articlesByIssue = $cache->get('articlesByIssue', $issueId);
+			}
+		}
+		return $articlesByIssue;
+	}
+
+	/**
+	 * Identify the issue of the given article.
+	 * @param $article PublishedArticle
+	 * @param $journal Journal
+	 * @return array|null Return prepared article data or
+	 *  null if the article is not from a published issue.
+	 */
+	function _prepareArticleData($article, $journal) {
+		$nullVar = null;
+
+		// Add the article to the cache.
+		$cache = $this->_getCache();
+		$cache->add($article, $nullVar);
+
+		// Retrieve the issue.
+		$issue = $this->_getArticleIssue($article, $journal);
+
+		if ($issue->getPublished()) {
+			$articleData = array(
+				'article' => $article,
+				'issue' => $issue
+			);
+			return $articleData;
+		} else {
+			return $nullVar;
+		}
+	}
+
+	/**
+	 * Return the object types supported by this plug-in.
+	 * @return array An array with object names and the
+	 *  corresponding export types.
+	 */
+	function _getAllObjectTypes() {
+		return array(
+			'issue' => DOAJ_EXPORT_ISSUES,
+			'article' => DOAJ_EXPORT_ARTICLES,
+		);
+	}
+
+	/**
+	 * Process a request.
+	 * @param $request PKPRequest
+	 * @param $journal Journal
+	 */
+	function _process($request, $journal) {
+		$objectTypes = $this->_getAllObjectTypes();
+		$target = $request->getUserVar('target');
```

Comment:
Mind trying to move this into the switch() as well? As you can tell, I'm not a big fan of assembling e.g. variable names using string operations.